### PR TITLE
feat: mention total limit

### LINKED ISSUE #810: Implement automod mention limits
### Summary

discord/discord-api-docs#5212, discord/discord-api-docs#5397

### What is the feature request for?

The core library

### The Problem

The above PRs are not implemented

### The Ideal Solution

- Add `mention_spam` to `AutoModerationTriggerType`.
- Add `mention_total_limit` to `AutoModerationTriggerMetadata`

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary

closes #810 

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -2909,6 +2909,10 @@ of :class:`enum.Enum`.
 
         This rule checks if content contains words from Discord pre-defined wordsets.
 
+    .. attribute:: mention_spam
+
+        This rule checks if the number of mentions in the message is more than the maximum allowed the number of mentions allowed.
```
Comment (DenverCoder1) [inline docs/api.rst:2914]: The wording here seems a bit repetitive
```suggestion
        This rule checks if the number of mentions in the message is more than the maximum allowed.
```

File: nextcord/auto_moderation.py [hunk 2]
```diff
@@ -95,20 +95,28 @@ class AutoModerationTriggerMetadata:
         .. warning::
 
             Wildcard syntax (`*`) is not supported here.
+    mention_total_limit: :class:`int`
+        The max amount of mentions allowed in one messsage.
```
Comment (DenverCoder1) [inline nextcord/auto_moderation.py:99]:
```suggestion
    mention_total_limit: Optional[:class:`int`]
        The maximum amount of mentions allowed in a messsage.
```

**Review by DenverCoder1 — COMMENTED** [review]
Just a couple small documentation suggestions, but otherwise looks good 👍

**Review by DenverCoder1 — APPROVED** [review]
Looks good, I haven't tested

**Review by teaishealthy — APPROVED** [review]

**Review by EmreTech — APPROVED** [review]
Haven't tested

File: docs/api.rst [hunk 3]
```diff
@@ -2909,6 +2909,10 @@ of :class:`enum.Enum`.
 
         This rule checks if content contains words from Discord pre-defined wordsets.
 
+    .. attribute:: mention_spam
+
+        This rule checks if the number of mentions in the message is more than the maximum allowed.
```
Comment (ooliver1) [inline docs/api.rst:2914]:
```suggestion
        This rule checks if the number of mentions in the message is more than the maximum allowed.

        .. versionadded:: 2.3
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/auto_moderation.py [hunk 4]
```diff
@@ -95,20 +95,28 @@ class AutoModerationTriggerMetadata:
         .. warning::
 
             Wildcard syntax (`*`) is not supported here.
+    mention_total_limit: Optional[:class:`int`]
+        The maximum amount of mentions allowed in a messsage.
+
```
Comment (ooliver1) [inline nextcord/auto_moderation.py:100]:
```suggestion

        .. versionadded:: 2.3

```

**Review by ooliver1 — APPROVED** [review]
Tested
```py
@bot.slash_command(name="create-rule", dm_permission=False)
async def create_rule(inter: Interaction):
    assert inter.guild is not None

    await inter.guild.create_auto_moderation_rule(
        name="Rule",
        trigger_type=AutoModerationTriggerType.mention_spam,
        trigger_metadata=AutoModerationTriggerMetadata(mention_total_limit=3),
        event_type=AutoModerationEventType.message_send,
        actions=[
            AutoModerationAction(
                type=AutoModerationActionType.send_alert_message,
                metadata=AutoModerationActionMetadata(
                    channel=inter.guild.get_channel(802586580766162967)
                ),
            )
        ],
        enabled=True,
    )
    await inter.send("ok")
```

**Pull request merged**